grafana-plugin/e2e-tests/labels/createNewLabelKeysAndValues.test.ts

```diff
@@ -0,0 +1,35 @@
+import { test, expect } from '../fixtures';
+import { clickButton, generateRandomValidLabel, openDropdown } from '../utils/forms';
+import { openCreateIntegrationModal } from '../utils/integrations';
+import { goToOnCallPage } from '../utils/navigation';
+
+test('New label keys and labels can be created', async ({ adminRolePage }) => {
+  const { page } = adminRolePage;
+  await goToOnCallPage(page, 'integrations');
+  await openCreateIntegrationModal(page);
+  const NEW_LABEL_KEY = generateRandomValidLabel();
+  const NEW_LABEL_VALUE = generateRandomValidLabel();
+
+  await page
+    .getByTestId('create-integration-modal')
+    .getByTestId('integration-display-name')
+    .filter({ hasText: 'Webhook' })
+    .first()
+    .click();
+  await clickButton({ page, buttonText: /^Add Labels$/ });
```

Comment:
Ad. 1 In this case we always create new integration and we always add the first key - value label pair so it will always be "Add Labels" here

Ad. 2 There is this eternal dilemma / discourse in e2e testing world which is: "Use test ids or explicit text-based locators" :D Both have pros and cons. Some years ago I would recommend to use test ids because it's the most resilient way. However:
- testing by ids is not user facing and placeholder / text can be important so using text-based locators is an expected part of assertion
- it adds some overhead because you need to maintain test ids and if you work with 3rd party libs (just as we work with grafana-ui) that don't support passing down test ids, you need to sometimes create additional markup to wrap target elem with some additional div with test id

Playwright team has some opinionated approach on that which is - prioritize role,label,text-based selectors over test ids. [Here](https://playwright.dev/docs/locators#quick-guide) you can see the list of locators and it's sorted by preference suggested by Playwright. Also the [test generator](https://playwright.dev/docs/codegen) or Locator tab in watch mode prioritizes text-based locators over test ids.

So imho if it's easy to select by text, we should freely use it and follow Playwright's opinion on that. If it's difficult or there is no visible text to use, then test-id is completely ok

Reply:
ok, thanks for such detailed explanation! In this case I think we need to keep using text-based locator here, and avoid using test-ids in the upcoming tests.